src/nvim/os/env.c

```diff
@@ -423,43 +423,41 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(const char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
-  int vimruntime;
-
-
-  p = (char_u *)os_getenv((char *)name);
-  if (p != NULL && *p == NUL)       /* empty is the same as not set */
-    p = NULL;
-
-  if (p != NULL) {
-    return p;
+  const char *kos_env_path = os_getenv(name);
+  if (kos_env_path != NULL
+      && *kos_env_path == NUL) /* empty is the same as not set */
+    kos_env_path = NULL;
+
+  if (kos_env_path != NULL) {
+    *mustfree = true;
+    return xstrdup(kos_env_path);
```

Comment:
This was all me.  If we return *p, then we could be returning a pointer to a value that is supposed to be a constant.  Rather than try to refactor everything around this function to treat the return value as a constant, I figured it'd be better to just return a copy of it, so it wouldn't have to be a constant -- but then the calling function needs to know it has to free that value right?